### PR TITLE
Rename BINARYEN_MEM_MAX to WASM_MEM_MAX

### PR DESCRIPTION
**Pull request by dschuff**
The name 'binaryen' is mostly an implementation detail of emscripten,
and the user-facing feature is actually wasm output.

Also introduces a settings-alias mechanism that we can use to canonicalize
flag names for backward compatibility.

### CODE REVIEW

**Comment by dschuff** [comment]
@kripken we talked about doing more of this renaming in the past, and I figured now would be a good time to fix BINARYEN_MEM_MAX before threads users start caring about it (today I'm guessing almost nobody uses it). What do you think of the settings-alias mechanism?

File: emcc.py [hunk 1]
```diff
@@ -688,6 +688,17 @@ def detect_fixed_language_mode(args):
             assert key != 'WASM_BACKEND', 'do not set -s WASM_BACKEND, instead set EMCC_WASM_BACKEND=1 in the environment'
       newargs = [arg for arg in newargs if arg is not '']
 
+      # Handle aliases in settings flags
+      settings_aliases = {
+          'BINARYEN_MEM_MAX': 'WASM_MEM_MAX',
```
Comment (kripken) [inline emcc.py]: maybe reorder these two lines, so WASM is before sub-WASM options?

**Comment by dschuff** [comment]
Yeah, I should add that switching the handling of Settings.BINARYEN -> WASM everywhere (including the aforementioned line 1100) is beyond the scope of this one PR, but if you're happy with the approach then I'll do that in a followup.

**Comment by jgravelle-google** [comment]
+1 to the general alias mechanism

**Comment by kripken** [comment]
Limiting the scope here sounds fine. So maybe just add a test for the change, and maybe remove the `WASM/BINARYEN` aliasing (since until a future PR it duplicates the code later down).

File: src/settings.js [hunk 2]
```diff
@@ -718,7 +718,7 @@ var BINARYEN_TRAP_MODE = "allow"; // How we handle wasm operations that may trap
 var BINARYEN_PASSES = ""; // A comma-separated list of passes to run in the binaryen optimizer,
                           // for example, "dce,precompute,vacuum".
                           // When set, this overrides the default passes we would normally run.
-var BINARYEN_MEM_MAX = -1; // Set the maximum size of memory in the wasm module (in bytes).
+var WASM_MEM_MAX = -1; // Set the maximum size of memory in the wasm module (in bytes).
                            // Without this, TOTAL_MEMORY is used (as it is used for the initial value),
```
Comment (kripken) [inline src/settings.js]: indentation broke

Comment (kripken) [reply src/settings.js]: also, maybe worth mentioning `(this was previously called "BINARYEN_MEM_MAX")`
